### ISSUE TITLE
SafeEdit dataset and implementation for MEND

### ISSUE THREAD

**Issue by lauraaisling** [zjunlp/EasyEdit#421]
I am looking to apply MEND model editing to remove toxicity with the SafeEdit dataset. 
I have seen in examples/run_safety_editing.md, it says in the example script that other methods are not available yet, but as MEND is a baseline in the paper, I guess it must be possible to apply MEND with SafeEdit. 

Firstly, is there an implementation available already? Secondly, is there a version of the SafeEdit dataset with the additional fields (such as rephrase)? (In issue 269 from a while back, https://github.com/zjunlp/EasyEdit/issues/269#issuecomment-2134235480 you say the dataset for mend is upcoming.)
In summary, could you please advise on how you implemented the MEND for SafeEdit as you did for table 1 of your SafeEdit paper "Detoxifying Large Language Models via Knowledge Editing"? 

Really appreciate any help or advice!

**Comment by mengrusun** [comment]
We have added MEND for SafeEdit.

**Comment by zxlzr** [comment]
hi, do you have any further issues?

**Comment by lauraaisling** [comment]
Hi @mengrusun and @zxlzr, looking at the MEND implementation, it makes sense to me, thanks for adding it and for responding so quick! I will test it out as soon as I can and let you know if I have any further questions. 
Cheers!

**Comment by ZHANGJINKUI** [comment]
> Hi @mengrusun and @zxlzr, looking at the MEND implementation, it makes sense to me, thanks for adding it and for responding so quick! I will test it out as soon as I can and let you know if I have any further questions. Cheers!

May I ask whether you have reproduced the MEND experiment of run_safety_editing.md? Why did the 48G card I used exceed the memory? If it is reproduced, could you share the specific parameter Settings? I would be very grateful!

**Comment by ABi-dot** [comment]
Same question: Traceback (most recent call last):
  File "/data3/lyf/research/EasyEdit/examples/run_safety_editing.py", line 229, in <module>
    trainer.run()
  File "/data3/lyf/research/EasyEdit/examples/../easyeditor/trainer/BaseTrainer.py", line 195, in run
    train_info = self.train_step(batch)
  File "/data3/lyf/research/EasyEdit/examples/../easyeditor/trainer/EditTrainer.py", line 141, in train_step
    l_total, l_edit, l_loc, l_base, info_dict = self.edit_step(
  File "/data3/lyf/research/EasyEdit/examples/../easyeditor/trainer/EditTrainer.py", line 48, in edit_step
    edited_model, model_info = self.model.edit(batch["edit_inner"])
  File "/data3/lyf/research/EasyEdit/examples/../easyeditor/trainer/algs/MEND.py", line 337, in edit
    loss.backward()
  File "/data3/lyf/miniconda3/envs/EasyEdit/lib/python3.9/site-packages/torch/_tensor.py", line 581, in backward
    torch.autograd.backward(
  File "/data3/lyf/miniconda3/envs/EasyEdit/lib/python3.9/site-packages/torch/autograd/__init__.py", line 347, in backward
    _engine_run_backward(
  File "/data3/lyf/miniconda3/envs/EasyEdit/lib/python3.9/site-packages/torch/autograd/graph.py", line 825, in _engine_run_backward
    return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 172.00 MiB. GPU 3 has a total capacity of 44.52 GiB of which 156.12 MiB is free. Including non-PyTorch memory, this process has 44.36 GiB memory in use. Of the allocated memory 43.59 GiB is allocated by PyTorch, and 431.93 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
